packages/edit-site/src/components/dataviews/view-actions.js

```diff
@@ -44,8 +44,17 @@ const availableViews = [
 	},
 ];
 
-function ViewTypeMenu( { view, onChangeView } ) {
-	const activeView = availableViews.find( ( v ) => view.type === v.id );
+function ViewTypeMenu( { view, onChangeView, supportedViewTypes } ) {
+	let _availableViews = availableViews;
+	if ( supportedViewTypes ) {
```

Comment:
What do you think of reusing the same array as in?

```js
if ( layouts ) {
  avaliableLayouts = availableLayouts.filter( ... );
}
```

An alternative would be to do the following name changes:

- `availableViews` => `layouts`
- `_availableViews` => `availableLayouts`

Reply:
I actually like the alternative naming best (layouts/availableLayouts).

Reply:
Ah, but then it conflicts with the prop. Perhaps?

```js
let availableLayouts = defaultLayouts;
if ( layouts ) {
  availableLayouts = defaultLayouts.filter( ... );
}
```

Reply:
Naming 😅

Reply:
We do have view type in some places style, we might need to think that properly. I'm not entirely sure "layout" is the right word in the "code" at least. For user facing things it's fine. But I kind of like the "block" / "block type" and "view" / "view type" parallel. 

Anyway, we need to make a decision and stick with it but it's probably one for a separate PR.